### PR TITLE
Review use of `db` fixture in tests

### PR DESCRIPTION
**Pull request by tcompa**
Ref:
* #588

### CODE REVIEW

File: tests/test_task_api.py [hunk 1]
```diff
@@ -36,7 +36,9 @@ async def test_task_get_list(db, client, task_factory, MockCurrentUser):
         assert data[2]["id"] == t2.id
 
 
-async def test_background_collection(db, dummy_task_package):
+async def test_background_collection(
+    db, registered_client, dummy_task_package
```
Comment (tcompa) [inline tests/test_task_api.py:40]: why registered client?

File: tests/test_workflow_api.py [hunk 2]
```diff
@@ -177,51 +196,42 @@ async def test_post_newtask(
...
 
-        workflow = await db.get(Workflow, wf_id)
-        t0 = await task_factory()
-        t1 = await task_factory()
-        await workflow.insert_task(t0.id, db=db)
-        await workflow.insert_task(t1.id, db=db)
-        await db.refresh(workflow)
-
-        assert len(workflow.task_list) == 2
-        assert workflow.task_list[0].task == t0
```
Comment (tcompa) [inline tests/test_workflow_api.py:188]: Missing assertion

File: tests/test_workflow_api.py [hunk 3]
```diff
@@ -247,10 +257,16 @@ async def test_delete_workflow_task(
...
+        t1 = add_task(client, 1)
+        t2 = add_task(client, 2)
+
+        payload = {"task_id": t1["id"]}  # FIXME
+        res = await client.post(
+            f"api/v1/workflow/{wf_id}/add-task/",
+            json=payload,
+        )
 
         await workflow.insert_task(t0.id, db=db)
```
Comment (tcompa) [inline tests/test_workflow_api.py:271]: Remove insert_task calls, and replace with three calls to add-task endpoint (as the one above)

Comment (tcompa) [reply tests/test_workflow_api.py:271]: Remove refresh

Comment (tcompa) [reply tests/test_workflow_api.py:271]: Probably
```python
        assert (
            len((await db.execute(select(WorkflowTask))).scalars().all()) == 3
            len((await db.execute(select(WorkflowTask))).scalars().all()) == 3
        )
```
will fail. If so, we can safely remove it

File: tests/test_workflow_api.py [hunk 4]
```diff
@@ -319,15 +335,16 @@ async def test_patch_workflow(
...
-        workflow = await db.get(Workflow, wf_id)
         res = await client.get(f"api/v1/project/{project.id}/workflows/")
         assert len(res.json()) == 1
 
         patch = {"name": "new_WF"}
         res = await client.patch(f"api/v1/workflow/{wf_id}", json=patch)
 
-        await db.refresh(workflow)
-        assert workflow.name == "new_WF"
+        res = await client.get(f"api/v1/workflow/{wf_id}")
```
Comment (tcompa) [inline tests/test_workflow_api.py:344]: use function

File: tests/test_workflow_api.py [hunk 5]
```diff
@@ -346,11 +363,17 @@ async def test_patch_workflow_task(
...
-        await workflow.insert_task(t0.id, db=db)
-        await db.refresh(workflow)
-        assert workflow.task_list[0].args is None
+        res = await client.get(f"api/v1/workflow/{wf_id}")
+        t = await add_task(client, 0)
+        payload = {"task_id": t["id"]}
+        res = await client.post(
+            f"api/v1/workflow/{wf_id}/add-task/",
+            json=payload,
+        )
```
Comment (tcompa) [inline tests/test_workflow_api.py:372]: Assert status code

File: tests/test_workflow_api.py [hunk 6]
```diff
@@ -395,23 +418,31 @@ async def test_patch_workflow_task_failures(
...
+
+        t2 = await add_task(client, 2)
+        res = await client.post(
+            f"api/v1/workflow/{wf2_id}/add-task/",
+            json={"task_id": t2["id"]},
+        )
+
+        res = await client.get(f"api/v1/workflow/{wf1_id}")
+        workflow1 = res.json()
+        res = await client.get(f"api/v1/workflow/{wf2_id}")
```
Comment (tcompa) [inline tests/test_workflow_api.py:436]: use function

File: tests/test_workflow_api.py [hunk 7]
```diff
@@ -535,7 +566,8 @@ async def test_reorder_task_list(
         workflow = {"name": "WF", "project_id": project.id}
         res = await client.post("api/v1/workflow/", json=workflow)
         wf_id = res.json()["id"]
-        workflow = await db.get(Workflow, wf_id)
+        res = await client.get(f"api/v1/workflow/{wf_id}")
```
Comment (tcompa) [inline tests/test_workflow_api.py:569]: use function

File: tests/test_task_api.py (hunk 1 above)
Comment (ychiucco) [reply tests/test_task_api.py:40]: Because with `client` we get `assert 401 == 200` at line 67 (unauthorized).

File: tests/test_task_api.py [hunk 8]
```diff
@@ -60,9 +62,11 @@ async def test_background_collection(db, dummy_task_package):
     await _background_collect_pip(
         state=state, venv_path=venv_path, task_pkg=task_pkg, db=db
     )
-    out_state = await db.get(State, state.id)
-    debug(out_state)
-    assert out_state.data["status"] == "OK"
+    res = await registered_client.get(f"{PREFIX}/collect/{state.id}")
```
Comment (tcompa) [inline tests/test_task_api.py:65]: Use with `MockCurrentUser`

File: tests/test_task_api.py (hunk 1 above)
Comment (tcompa) [reply tests/test_task_api.py:40]: Use with MockCurrentUser

**Pull request merged**